### ISSUE TITLE
Can't load test data with DataBunch

### ISSUE THREAD

**Issue by jfilter** [fastai/fastai#918]
**Describe the bug**

Can't load test dataset with `DataBunch.from_folder`.

**To Reproduce**
Create three folders: 'train', 'valid', and 'test' with folder for the classes and images in those folders. Then try to run the following:

```
data = ImageDataBunch.from_folder(PATH, test='test')

learn = ConvLearner(data, tvm.resnet18, metrics=accuracy)
learn.fit(1)

preds,y = learn.get_preds(is_test=True)
```

**Expected behavior**
I want to test the performance of my model on the test datasets.

**Comment by jfilter** [comment]
Hey, thanks for your help.

I am happy to provide more information, but I think we found the problem. My images in `test` are in sub-folders (for the classes) similar to the training and validation folders.

I don't understand your concept of 'test set' in the library. Normally, you indeed have the labels for the test set. [Some background about the split into train, val and test](https://stats.stackexchange.com/questions/19048/what-is-the-difference-between-test-set-and-validation-set)

Does fastai mean 'predict' when you 'test'?

**Comment by sgugger** [comment]
Yes, the test set is intended for predictions, the base example being a Kaggle competition where you don't have the labels. If you have a second test set with labels, you should can validate your model on it by using `learn.validate(test_dl)`, `test_dl` being a dataloader that will iterate through it.

**Comment by jfilter** [comment]
Thanks for the clarifications.
